src/main/scala/com/scality/clueso/MetadataIngestionPipeline.scala

```diff
@@ -0,0 +1,187 @@
+package com.scality.clueso
+
+import com.fasterxml.jackson.databind.node.{ArrayNode, NullNode, ObjectNode, TextNode}
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.streaming.{OutputMode, ProcessingTime}
+import org.apache.spark.sql.types.TimestampType
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import scala.collection.mutable
+
+class EventMessageRewriter {
+  val mapper = new ObjectMapper()
+
+  def rewriteMsg(bytes: Array[Byte]): String = {
+    val payload = new String(bytes)
+
+    var rootNode = mapper.readTree(payload).asInstanceOf[ObjectNode]
+
+    val valueTxtNode = rootNode.path("value").asInstanceOf[TextNode]
```

Comment:
why do you need this step?

Reply:
To parse the value as a string. That string is then JSON parsed again.

Reply:
will you please add a comment that this needs to be done since the value is double json stringified

Reply:
(y)